skyportal/handlers/api/spectrum.py

```diff
@@ -140,6 +140,11 @@ def post(self):
             payload={'obj_key': spec.obj.internal_key},
         )
 
+        self.push_all(
```

Comment:
This should either just be `push` (instead of `push_all`) or you need to add a check to the WS message handler to see whether the spectra for the relevant obj is even loaded for each user, otherwise you're triggering a ton of unnecessary fetches here

Reply:
This needs to be push all as this ensures that the deletion is responsively propagated to all users looking at the spectrum page

Reply:
Then the front-end WS message handler needs to check whether the user is actually viewing the relevant spectra page, and not just trigger a fetch for everyone

Reply:
fixed. thanks for catching this subtle issue